src/Symfony/Component/Process/Process.php

```diff
@@ -1455,6 +1482,11 @@ private function resetProcessData(): void
      */
     private function doSignal(int $signal, bool $throwException): bool
     {
+        // Signal seems to be send when sigchild is enable, this allow blocking the signal correctly in this case
+        if ($this->isSigchildEnabled() && \in_array($signal, $this->ignoredSignals)) {
```

Comment:
what if we keep only this logic to not have to rely on pcntl_sigprocmask, which is not portable?

Reply:
This logic only work for signal sent in PHP, pcntl_sigprocmask allow to make it work also when a signal is received by the PHP binary (fpm / cli / etc ...) which is the initial case for this feature: to not send SIGTERM to underlying process create by this component (which is done when doing a systemctl stop on a service that is a php symfony command)

This last case seems to also work when php is compiled with --enable-sigchild, i try to make a test to better have this behavior but it's quite hard, (i think i need to fork and detach a process for that, otherwise current fork behavior mess up with what's intended) but not sure if we want to do that in a test ?

Reply:
got it thanks